tempodb/encoding/blocker.go

```diff
@@ -0,0 +1,51 @@
+package encoding
+
+import (
+	"context"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type Finder interface {
+	FindTraceByID(ctx context.Context, id common.ID) (*tempopb.Trace, error)
+}
+
+type Searcher interface {
+	Search(ctx context.Context, req *tempopb.SearchRequest, opts ...SearchOption) (*tempopb.SearchResponse, error)
+}
+
+type SearchOptions struct {
+	chunkSizeBytes     uint32
+	startPage          int
+	totalPages         int
+	maxBytes           int
+	prefetchTraceCount int
+}
+
+type SearchOption func(*SearchOptions)
+
+func WithPages(startPage, totalPages int) SearchOption {
```

Comment:
I, personally, am not a fan of the functional options pattern. I think it makes it harder to interact with the struct, but I'm also not adamantly opposed to it. 

I don't believe we use this anywhere else, but if the team wants to move this direction I am ok with it.

Reply:
Team slightly leaned against this pattern, so replaced it with a basic struct.